pkg/controller/networkaddonsconfig/networkaddonsconfig_controller.go

```diff
@@ -195,7 +195,7 @@ func (r *ReconcileNetworkAddonsConfig) Reconcile(request reconcile.Request) (rec
 	}
 
 	// Canonicalize and validate NetworkAddonsConfig, finally render objects of requested components
-	objs, err := r.renderObjects(networkAddonsConfig)
+	objs, objsToRemove, err := r.renderObjects(networkAddonsConfig)
```

Comment:
Again, even here we should call to r.objectsToDelete, render is meant only to interface call to the Render.

Reply:
yep this already did , but didnt push yet, sorry for the confusing

Reply:
but renderObjects calculate much stuff that is needed for both network.Render and network.RenderObjsToRemove
its bad to do them twice i think ? i will push some commits and lets respin thinking ?

Reply:
but renderObjects calculate much stuff that is needed for both network.Render and network.RenderObjsToRemove
its bad to do them twice i think ? i will push some commits and lets respin thinking ?

Reply:
It does a lot of stuff that we absolutely don't need for the removal. Let's start by having a separate function for it and then we can see if we can make a parent function both of those to cover the shared parts.

It would make it cleared for the review to see what information you need to perform the removal. And when do you gather it.

Reply:
reread now all the stuff it does before render , i think i need all, but i will split to two func and make the common part as a func and then we can review that common func if i am not wrong that we need all of it

(the suffix is just needed for render itself)